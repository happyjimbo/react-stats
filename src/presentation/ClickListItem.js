import React, {PropTypes, Component} from 'react';
import {ListGroupItem} from 'react-bootstrap';

class ClickListItem extends Component {

    render() {

       let statName = this.props.statName;
       let info = this.props.info;
       let click = this.props.click;
       let StatsLabelContainer = this.props.StatsLabelContainer;

       let style = this.props.success ? "success" : "danger";

        return (
            <ListGroupItem onClick={click} bsStyle={style}>
                <b>{statName}</b> : {info}
            </ListGroupItem>
        ); 
    }

    static propTypes = {
        click: PropTypes.func.isRequired
    }
}

export default ClickListItem;