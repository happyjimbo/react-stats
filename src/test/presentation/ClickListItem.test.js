import React from 'react';
import {shallow} from 'enzyme';
import ClickListItem from '../../presentation/ClickListItem';
import renderer from 'react-test-renderer';

describe('ClickListItem', () => {

    const statName = "MyStatName";
    const info = "some info";
    const click = () => {};

    function setUp(extraProps = undefined) {

        const defaultProps = {
            statName,
            info,
            click
        };

        const props = Object.assign({}, defaultProps, extraProps);
        const component = shallow(<ClickListItem {...props} />);

        return {
            props,
            component
        }
    }

    it('compares component to snapshot', () => {
        
        const style = true;

        const component = renderer.create(
            <ClickListItem statName={statName} info={info} click={click} style={style} />
        );

        let clickListItem = component.toJSON();
        expect(clickListItem).toMatchSnapshot();
    });

    it ('runs click callback when clicked, setting value to true', () => {

        let clicked = false;
        const click = () => { clicked = true; };

        const props = { click };
        const { component } = setUp(props);
        
        expect(clicked).toBe(false);

        component.simulate('click');
        
        expect(clicked).toBe(true);
    });

    it ('sets byStyle to success state', () => {

        const props = { success: true };
        const { component } = setUp(props);

        const componentProps = component.props();
        expect(componentProps.bsStyle).toBe("success");
    });

    it ('sets byStyle to danger state', () => {

        const props = { success: false };
        const { component } = setUp(props);

        const componentProps = component.props();
        expect(componentProps.bsStyle).toBe("danger");
    });

    it ('displays statName and info', () => {

        const { component } = setUp();

        const containsStatName = component.children().contains(statName);
        expect(containsStatName).toBe(true);

        const containsInfo = component.children().contains(info);
        expect(containsInfo).toBe(true);
    });
    
});