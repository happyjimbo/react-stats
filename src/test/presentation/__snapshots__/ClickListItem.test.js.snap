exports[`ClickListItem compares component to snapshot 1`] = `
<button
  className="list-group-item list-group-item-danger"
  onClick={[Function]}
  type="button">
  <b>
    MyStatName
  </b>
   : 
  some info
</button>
`;
